### DISCUSSION TITLE
I'm having trouble with the api server authorization...

### DISCUSSION THREAD

**Discussion by OGbilboswagins** [pear-devs/pear-desktop#2797] [Q&A] [answered]
I apologize ahead of time for my complete ignorance on the subject and wasting your time with a potentially stupid question. I have managed to find the correct url using the /swagger endpoint, but I can't for the life of me figure out how to generate a bearer access token. I have searched the interwebs high and low, looked over the /doc, scoured the "Swagger" manual, all for the last 3 hours trying to find the answer but have failed. Would someone be so kind as to explain to me the steps I need to take to generate a bearer authorization token or point me to some relevant documentation on the subject? I don't even know where to start. Thank you so much in advance.

**Comment by kimjammer** [comment, marked as answer]
1. Send a POST request to /auth/{id}, where {id} is the name of your plugin (it's basically only displayed in the authorization popup, it doesn't really matter.)
2. Click Authorize in the youtube-music popup (might be hidden behind other windows)
3. The response from the POST request will contain the accessToken field. This is your bearer access token.
4. In all future requests to the API, in the header, include an "Authorization" field with the value set to "Bearer your-token-here"
The swagger page has a "try it out" button, I highly recommend playing around with it to learn how it works.
Example auth request
![image](https://github.com/user-attachments/assets/968a11f5-a128-4a49-835b-cfd180d133ed)
Example api request
![image](https://github.com/user-attachments/assets/7ccba0be-4c2f-4ebf-ba37-d9e24d961d14)
PS: Sometimes the auth is glitchy and you need to restart the app before your token works.
